Structured MVP with backend layers

Kalyan0508 opened the pull request:
Features Implemented:

Backend setup using Node.js + Express

Integrated PostgreSQL for metadata storage

Enabled local image upload using Multer

Created uploads/ directory for storing images on disk

Built two REST APIs:

POST /api/upload: Upload image with metadata (user, location, timestamp)

GET /api/images: Fetch list of uploaded images with metadata

Configured environment variables using .env

API tested successfully using Postman (form-data)

Structured code with controllers, routes, and db folders

Tech Stack:
Node.js + Express

PostgreSQL

Multer for file handling

dotenv for env configs

*************************
PI Upload and Fetch Test via Postman
POST /api/upload
Uploaded an image along with user metadata.

Status: 201 Created

Response:
{
"message": "Image uploaded successfully"
}
postman-POST
![postman-POST](https://github.com/user-attachments/assets/9c272815-22da-4c25-acf2-aa501e03c769)


GET /api/images
Retrieved uploaded image data and metadata.

Status: 200 OK

Response:
[
{
"id": 2,
"user_name": "John",
"location": "hyderabad",
"image_path": "C:\Users\...\uploads\17471....jpg",
"timestamp": "2025-05-14T04:35:15.047Z"
},
{
"id": 1,
"user_name": "Kalyan Ram",
"location": "Khammam",
"image_path": "uploads\17471....jpg",
"timestamp": "2025-05-13T16:29:25.410Z"
}
]

![postman-GET](https://github.com/user-attachments/assets/94644efb-2aee-455e-a8a9-7585327a4942)

Kalyan0508 commented:
SIr ,Now I hope it resolved  if not kindly inform me.

cpandey05 commented:
So many files?lib, node modules these should never go to repo .. due to huge size i am unable to add file level comment, bur your service logic is mixed up with controller and do not see any repo layer!

Kalyan0508 commented:
I've pushed the requested changes, including folder cleanup and restructuring as suggested. Let me know if anything else needs updating!

Trainings-TechEazy commented on repositories/imageRepository.js:
```diff
@@ -0,0 +1,13 @@
+const db = require('../db/db');
+
+exports.insertImage = (user, location, imagePath, timestamp) => {
+  return db.query(
```
What if user decides to send 3 more params

Trainings-TechEazy commented on repositories/imageRepository.js:
```diff
@@ -0,0 +1,13 @@
+
+exports.getAllImages = async () => {
+  const result = await db.query("SELECT * FROM images ORDER BY timestamp DESC");
+  return result.rows;
```
Why all properties

Kalyan0508 commented:
1.Refactored the repository to dynamically handle any number of fields during image insert.

2.Changed SELECT * to select only required fields (id, user_name, location, image_path, timestamp) to optimize queries.

3.Added error checks and improved code comments for clarity.

sir and also a database connection pool is a collection of unused database connections that are already open.

Now kindly inform me for some other changes and its my pleasure that you are giving me suggestions for improvement

cpandey05 commented:
I do not see your attendance in any of the scrum?
